### PR TITLE
refactor(comment): port to Media layout

### PR DESCRIPTION
**Pull request by alexkrolick**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @alexkrolick has signed the CLA. Thanks for the pull request!

File: src/elements/content-sidebar/activity-feed/comment/Comment.js [hunk 1]
```diff
@@ -66,7 +67,7 @@ class Comment extends React.Component<Props, State> {
 
     state = {
         isConfirmingDelete: false,
-        isEditing: false,
+        isEditingInline: false,
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/Comment.js]: Is this change related to the Media component refactor?

File: src/elements/content-sidebar/activity-feed/comment/Comment.js [hunk 2]
```diff
@@ -129,22 +130,52 @@ class Comment extends React.Component<Props, State> {
             getMentionWithQuery,
             mentionSelectorContacts,
         } = this.props;
-        const { isConfirmingDelete, isEditing, isInputOpen } = this.state;
+        const { isConfirmingDelete, isEditingInline, isInputOpen } = this.state;
         const createdAtTimestamp = new Date(created_at).getTime();
         const createdByUser = created_by || PLACEHOLDER_USER;
+        const isTask = type === COMMENT_TYPE_TASK;
+        const isEditSupported = isTask; // comment editing not supported
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/Comment.js]: We usually try to avoid rename-only reassignments. Can we use `isTask` where it's needed?

File: src/elements/content-sidebar/activity-feed/comment/Comment.js [hunk 3]
```diff
@@ -129,22 +130,52 @@ class Comment extends React.Component<Props, State> {
...
+        const canEdit = permissions.can_edit && isEditSupported;
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/Comment.js]: Looks like these can be destructured/renamed from `permissions`?

File: src/elements/content-sidebar/activity-feed/comment/__tests__/CommentDeleteConfirmation-test.js [hunk 4]
```diff
@@ -42,19 +42,5 @@ describe('elements/content-sidebar/ActivityFeed/comment/CommentDeleteConfirmatio
             expect(onDeleteCancelMock).toBeCalled();
             expect(onDeleteConfirmMock).not.toBeCalled();
         });
-
-        test('should handle Enter key', () => {
```
Comment (alexkrolick) [inline src/elements/content-sidebar/activity-feed/comment/__tests__/CommentDeleteConfirmation-test.js]: this was a bug - it autofocuses the cancel button so enter should just trigger the default handler

File: src/elements/content-sidebar/activity-feed/comment/CommentDeleteConfirmation.js [hunk 5]
```diff
@@ -36,13 +37,6 @@ class CommentDeleteConfirmation extends React.Component<Props> {
                     onDeleteCancel();
                 }
                 break;
-            case KEYS.enter:
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/CommentDeleteConfirmation.js]: Is this no longer needed?

File: src/elements/content-sidebar/activity-feed/comment/CommentDeleteConfirmation.scss [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    margin-bottom: 20px;
+}
+
+.bcs-CommentDeleteConfirmation-delete,
+.bcs-CommentDeleteConfirmation-cancel {
+    margin-left: 0;
+}
+
+.bcs-CommentDeleteConfirmation-cancel {
+    margin-right: 10px;
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/CommentDeleteConfirmation.scss]: Can we swap this for `margin-left` on the delete button?

File: src/elements/content-sidebar/activity-feed/comment/CommentMenuItems.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+const CommentMenuItems = (props: Props) => {
+    const { onDeleteClick, onEditClick, permissions = {}, type } = props;
+    const isTask = type === COMMENT_TYPE_TASK;
+    const isEditSupported = isTask; // comment editing not supported
+    const editLabel = isTask ? messages.taskEditMenuItem : commonMessages.editLabel;
+    const deleteLabel = isTask ? messages.taskDeleteMenuItem : deleteMessages.deleteLabel;
+
+    return (
+        <>
+            {!!onEditClick && !!permissions.can_edit && isEditSupported && (
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/CommentMenuItems.js]: Can we pass in `canEdit` and `canDelete` from the parent rather than `permissions`? It looks to be accessed in both places, currently.

File: src/elements/content-sidebar/activity-feed/comment/CommentMenuItems.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+import commonMessages from '../../../common/messages';
+import messages from './messages';
+import deleteMessages from '../inline-delete/messages';
+import { ACTIVITY_TARGETS } from '../../../common/interactionTargets';
+import { bdlGray80 } from '../../../../styles/variables';
+import { COMMENT_TYPE_DEFAULT, COMMENT_TYPE_TASK } from '../../../../constants';
+
+type Props = {
+    id: string,
+    onDeleteClick?: Function,
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/CommentMenuItems.js]: More specific types?

File: src/elements/content-sidebar/activity-feed/comment/__tests__/CommentDeleteConfirmation-test.js (hunk 4 above)
Comment (booc0mtaco) [reply src/elements/content-sidebar/activity-feed/comment/__tests__/CommentDeleteConfirmation-test.js]: Had to confirm for myself since I think i added this. `<enter>` on a non-submit cancel button should do the `onclick` (but `<enter>` in a field in a form should do the `onsubmit` of the form.) This was just doing the action of the delete button

https://jsfiddle.net/rs17ep2z/

File: src/elements/content-sidebar/activity-feed/comment/CommentDeleteConfirmation.scss (hunk 6 above)
Comment (alexkrolick) [reply src/elements/content-sidebar/activity-feed/comment/CommentDeleteConfirmation.scss]: no, when the 2 buttons wrap, the right button would not be left-aligned

File: src/elements/content-sidebar/activity-feed/comment/Comment.js [hunk 9]
```diff
@@ -132,19 +138,65 @@ class Comment extends React.Component<Props, State> {
...
+                                className="bcs-comment-delete-confirm"
+                                constraints={[{ to: 'scrollParent', attachment: 'together' }]}
+                                targetAttachment="bottom right"
+                            >
+                                <Media.Menu isDisabled={isConfirmingDelete} data-testid="open-actions-menu">
+                                    {canEdit && isTask && (
+                                        <MenuItem
+                                            className="bcs-comment-menu-edit"
+                                            data-resin-target={ACTIVITY_TARGETS.INLINE_EDIT}
+                                            onClick={this.handleDeleteClick}
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/Comment.js]: Did the handlers get swapped?

File: src/elements/content-sidebar/activity-feed/comment/Comment.js [hunk 10]
```diff
@@ -132,19 +138,65 @@ class Comment extends React.Component<Props, State> {
...
+                                        >
+                                            <IconPencil color={bdlGray80} />
+                                            <FormattedMessage
+                                                {...(isTask ? messages.taskEditMenuItem : commonMessages.editLabel)}
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/comment/Comment.js]: The parent won't be displayed at all if `isTask` is false, right?

**Comment by jstoffan** [comment]
Looks good. Could you post a before/after screenshot of both comments and tasks, please?

File: src/elements/content-sidebar/activity-feed/comment/Comment.js (hunk 9 above)
Comment (alexkrolick) [reply src/elements/content-sidebar/activity-feed/comment/Comment.js]: 😱 as a followup writing dom tests would be good

File: src/elements/content-sidebar/activity-feed/comment/Comment.js (hunk 10 above)
Comment (alexkrolick) [reply src/elements/content-sidebar/activity-feed/comment/Comment.js]: true... the shortcircuit is a bit tacked-on

Comment (alexkrolick) [reply src/elements/content-sidebar/activity-feed/comment/Comment.js]: Actually, I will pull up the `isTask` into the `canEdit` variable. Legacy tasks use inline edit so while that code is still here this should be supported.

**Comment by alexkrolick** [comment]
**New**

Screen Shot 2019-07-17 at 4 16 37 PM


**Old**

old-comment-card

**Comment by booc0mtaco** [comment]
Prob doesn't matter **too** much but looks like the color of the ellipsis changed to something a bit darker. intentional?

File: src/elements/content-sidebar/activity-feed/task-new/Task.scss [hunk 11]
```diff
@@ -39,7 +26,7 @@ $bcs-task-left-offset: 10 * $bcs-task-grid;
     @include avatar-badge-icon;
 
     border-width: 2px;
-    right: $bcs-task-grid;
+    right: -4px;
```
Comment (booc0mtaco) [inline src/elements/content-sidebar/activity-feed/task-new/Task.scss]: `-$bcs-task-grid` ?

Comment (alexkrolick) [reply src/elements/content-sidebar/activity-feed/task-new/Task.scss]: as far as I can tell the relationship between the grid and the avatar offset was incidental, it's actually based on the border width

**Comment by alexkrolick** [comment]
@booc0mtaco the ellipsis was `nines` so we can either go lighter to `bdl-gray-50` (#9a9a9a) or darker to the min recommended lightness for WCAG2.0 `bdl-gray-62`)

File: src/elements/content-sidebar/activity-feed/task-new/Task.scss (hunk 11 above)
Comment (booc0mtaco) [reply src/elements/content-sidebar/activity-feed/task-new/Task.scss]: ah, it actually comes from that `border-width` value then 👍
